### PR TITLE
DROOLS-2530 Wrong evaluation of parameter named "product"

### PR DESCRIPTION
**Pull request by kurobako**
https://issues.jboss.org/browse/DROOLS-2530

### CODE REVIEW

**Comment by kurobako** [comment]
@tarilabs I think we might want to additionally log warnings in case some name from the model shadows built-in function like 'product' in this case. What do you think?

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/impl/EvaluationContextImpl.java [hunk 1]
```diff
@@ -140,8 +141,9 @@ public boolean isDefined(String[] name) {
     @Override
     public Map<String, Object> getAllValues() {
         Map<String, Object> values = new HashMap<>(  );
-        for (ExecutionFrame frame : stack) {
-            values.putAll( frame.getAllValues() );
+        Iterator<ExecutionFrame> it = stack.descendingIterator();
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/impl/EvaluationContextImpl.java:144]: implementing reversing order as expected. thanks.

**Review by tarilabs — APPROVED** [review]
Unfortunately I know @etirelli view on this kind of features is that such additional check would hit performance quite heavily (thinking about how many times this method is called).
So, I would refrain from introducing this check for now.

However, you are suggesting a feature which might be possible to some extent in the validation. I suggest you raise a new JIRA as "enhancement" (ie: not bug) and we discuss this potential check later in more details.

What do you think?

Besides this correction looks consistent, thanks!

**Comment by tarilabs** [comment]
@baldimir didn't realize yet you were not tagged for review, for me is fine, unless there are further comments I'll ask Mario to merge.

**Review by baldimir — APPROVED** [review]
Looks ok, thanks.

**Pull request merged**